src/hyp3_sdk/jobs.py

```diff
@@ -180,14 +183,16 @@ def __len__(self):
     def __contains__(self, job: Job):
         return job in self.jobs
 
-    def __eq__(self, other: 'Batch'):
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Batch):
+            raise NotImplementedError('`__eq__` can only compare a Batch object with another Batch object.')
```

Comment:
`TypeError` is what is raised if I do e.g. `'foo' + 1`
```suggestion
            raise TypeError('`__eq__` can only compare a Batch object with another Batch object.')
```